### ISSUE TITLE
Android BOINC 7.16.16 does not stop computations immediately when being disconnected from a charger

### ISSUE THREAD

**Issue by Ageless93** [BOINC/boinc#4192]
When I unplug the phone, I expect tasks that are running to suspend within moments, maybe a maximum of 10 seconds for BOINC to write their state to the SD. But instead the running tasks show as "suspending" for over a minute while they keep increasing in percentage, thus being calculated, while on battery. 

https://ibb.co/Dk3Ct0f
https://ibb.co/4J2L54Q

**Comment by Ageless93** [comment]
Okay, so after much much testing this is a problem with the Universe tasks. These will at all times do Suspending for at least a minute if you keep BOINC open. But I don't always have BOINC open when I just detach the phone. And here's where it gets worse, these tasks now stay in memory while Suspending until the BOINC client crashes & restarts. 

Or it shown in images:
https://i.ibb.co/GQDd5p3/Suspending-tasks-don-t-suspend-when-not-on-charger.jpg
https://i.ibb.co/RjBzMqp/2-minutes-later.jpg
https://i.ibb.co/DKXbKrT/Universe-tasks-now-show-Ready.jpg
https://i.ibb.co/Y0Nm0WZ/Because-the-client-crashed-and-restarted-at-1538.jpg

When I leave BOINC open for more than a minute after disconnecting it from the charger, BOINC will quit the tasks with a forced kill command. It doesn't do so when BOINC isn't open and it's disconnected. I don't have a quick nice solution for this.

**Comment by Ageless93** [comment]
Oh, the client here crashed only because I unlocked my phone around that time. I saw a flurry of notifications from BOINC, switching rapidly from Calculating to Suspended even though it wasn't on a charger. I have noticed that when I do not unlock my phone in this state and just reconnect it to a charger, that these tasks will sit in perpetuity waiting to suspend. The client won't crash & restart either. The tasks won't suspend further and restart into Running mode. They just won't do anything and with that hold up all other tasks waiting to run as well. 

It's only when I here unlock the phone, open BOINC and leave it for over a minute on the Tasks tab, that these tasks will either be force killed by BOINC and restarted, or that the client crashes and restarts.
